framework/include/problems/EigenProblem.h

```diff
@@ -275,6 +269,26 @@ class EigenProblem : public FEProblemBase
   bool & _first_solve;
   /// A value used for initial normalization
   Real _initial_eigenvalue;
+
+private:
+  /**
+   * Do some free/extra power iterations
+   */
+  void doFreeNonlinearPowerIterations(unsigned int free_power_iterations);
+
+  /**
+   * Adjust eigen vector by either scaling the existing values or setting new values
+   * The operations are applied for only eigen variables
+   */
+  void adjustEigenVector(const Real value, bool scaling);
+
+  /// The name of the Postprocessor providing the Bx norm. This may be empty in which case the
+  /// default L2 norm of Bx will be used as the Bx norm
+  const PostprocessorName _bx_norm_name;
```

Comment:
If you care (this is not required) - cool opportunity to get used to `std::optional`. Could have a `const std::optional<PostprocessorName> _bx_norm_name`, and then your logic later on is just `operator bool` for `_bx_norm_name`